src/rta/fullsupernodelist.cpp

```diff
@@ -239,40 +239,74 @@ bool FullSupernodeList::buildAuthSample(uint64_t height, vector<SupernodePtr> &o
     }
 
     epee::string_tools::hex_to_pod(block_hash_str, block_hash);
-    vector<SupernodePtr> tier_supernodes;
-    out.clear();
-
-    auto out_it = back_inserter(out);
-
-
-    auto build_tier_sample = [&](uint64_t tier_min, uint64_t tier_max, size_t max_items) -> size_t {
-        if (max_items == 0)
-            return 0;
-        selectTierSupernodes(block_hash, tier_min, tier_max, tier_supernodes, out, max_items);
-        copy(tier_supernodes.begin(), tier_supernodes.end(), out_it);
-        size_t items_selected =  tier_supernodes.size();
-        tier_supernodes.clear();
-        return items_selected;
-    };
 
-    // infinite loop protection
-    size_t tries_left = AUTH_SAMPLE_SIZE * AUTH_SAMPLE_SIZE; // not sure if it needs to be N^2
+    std::array<std::vector<SupernodePtr>, TIERS> tier_supernodes;
```

Comment:
looks like splitting the code bellow into multiple methods would significantly improve its readability